### PR TITLE
Remove a lot of dead code

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: src/main/java/org/truffleruby/parser/TranslatorEnvironment.java [hunk 1]
```diff
@@ -172,7 +168,6 @@ public ReadLocalNode findLocalVarNode(String name, Source source, SourceIndexLen
             if (current != null) {
                 current = this;
                 while (level-- > 0) {
-                    current.needsDeclarationFrame = true;
                     current = current.parent;
                 }
             }
```
Comment (eregon) [inline src/main/java/org/truffleruby/parser/TranslatorEnvironment.java]: The whole `finally` clause can be removed.

File: src/main/java/org/truffleruby/parser/RubyWarnings.java [hunk 2]
```diff
@@ -147,12 +70,9 @@ public void warn(ID id, String fileName, int lineNumber, String message) {
         buffer.append("warning: ").append(message).append('\n');
         DebugHelpers.eval(runtime, "$stderr.write Truffle::Interop.from_java_string(message)", "message", buffer.toString());
     }
-
-    /**
-     * Prints a warning, unless $VERBOSE is nil.
-     */
```
Comment (eregon) [inline src/main/java/org/truffleruby/parser/RubyWarnings.java]: Please keep the javadoc and add it the `warn` above too.
It means we don't need to look at the source to know what it does.

File: src/main/java/org/truffleruby/parser/parser/RubyParser.java [hunk 3]
```diff
@@ -3627,7 +3627,7 @@ public Object yyparse (RubyLexer yyLex) {
 };
 states[458] = new ParserState() {
   @Override public Object execute(ParserSupport support, RubyLexer lexer, Object yyVal, Object[] yyVals, int yyTop) {
-                    yyVal = support.literal_concat(((ParseNode)yyVals[-1+yyTop]).getPosition(), ((ParseNode)yyVals[-1+yyTop]), ((ParseNode)yyVals[0+yyTop]));
+                    yyVal = support.literal_concat(((ParseNode)yyVals[-1+yyTop]), ((ParseNode)yyVals[0+yyTop]));
```
Comment (eregon) [inline src/main/java/org/truffleruby/parser/parser/RubyParser.java]: All changes to RubyParser.java need to be reflected in the `.y`.

**Comment by eregon** [comment]
@chrisseaton How did you find all this code? IDE or manually?

File: src/main/java/org/truffleruby/parser/TranslatorEnvironment.java (hunk 1 above)
Comment (chrisseaton) [inline src/main/java/org/truffleruby/parser/TranslatorEnvironment.java]: Done.

File: src/main/java/org/truffleruby/parser/RubyWarnings.java (hunk 2 above)
Comment (chrisseaton) [inline src/main/java/org/truffleruby/parser/RubyWarnings.java]: Done.

File: src/main/java/org/truffleruby/parser/parser/RubyParser.java (hunk 3 above)
Comment (chrisseaton) [inline src/main/java/org/truffleruby/parser/parser/RubyParser.java]: Done.

**Comment by chrisseaton** [comment]
IntelliJ highlights dead code (does Eclipse not?) but then some manual unpicking usually reveals more complex dead code.

File: src/main/java/org/truffleruby/parser/parser/RubyParser.java (hunk 3 above)
Comment (eregon) [inline src/main/java/org/truffleruby/parser/parser/RubyParser.java]: Thanks, #285 will automatically check that in the future.

**Comment by eregon** [comment]
Eclipse does not at least with default settings (but maybe it's a good thing too, I sometimes find IntelliJ overwhelming with how much it tries to tell you on every few lines).

**Pull request merged**